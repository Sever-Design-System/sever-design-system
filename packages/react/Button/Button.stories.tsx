import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Button } from './Button';
import { PlusIcon } from '../Icons';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Button',
  component: Button,
} as ComponentMeta<typeof Button>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Primary.args = {
  label: 'Button',
  size: 'medium',
  appearance: 'primary',
};

export const WithLeftIcon = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
WithLeftIcon.args = {
  label: 'Button',
  size: 'medium',
  appearance: 'primary',
  IconLeft: PlusIcon(),
};

export const WithRightIcon = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
WithRightIcon.args = {
  label: 'Button',
  size: 'medium',
  appearance: 'primary',
  IconLeft: PlusIcon(),
};

export const Loading = Template.bind({});
WithRightIcon.args = {
  label: 'Button',
  size: 'medium',
  appearance: 'primary',
  isLoading: true,
};
