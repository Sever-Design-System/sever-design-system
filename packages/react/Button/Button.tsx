import { FC } from 'react';

export interface ButtonProps {
  /**
   * Button contents
   */
  label: string;
}

/**
 * Primary UI component for user interaction
 */
const Button: FC<ButtonProps> = ({
  label,
  ...props
}) => {
  return (
    <button
      type="button"
      {...props}
    >
      {label}
    </button>
  );
};






export default Button;
