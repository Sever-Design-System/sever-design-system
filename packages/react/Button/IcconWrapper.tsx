import { styled } from '../stitches.config';

export const IconWrapper = styled('span', {
  display: 'inline-flex',
  alignSelf: 'center',

  variants: {
    left: {
      true: {
        marginInlineEnd: '$xs',
      },
    },
    right: {
      true: {
        marginInlineStart: '$xs',
      },
    },
  },
});
