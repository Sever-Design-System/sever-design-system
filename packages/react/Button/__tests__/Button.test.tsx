import * as renderer from 'react-test-renderer';
import { Button } from '../Button';

// const propsToTest = {
//   label: 'Some button text',
//   view: ['outline', 'default', 'link', 'text'],
//   size: ['small', 'medium', 'large'],
// };
describe('Button component testing', () => {
  test('Button props', () => {
    const component = renderer.create(
      <Button size="medium" appearance="primary" label="Button" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
