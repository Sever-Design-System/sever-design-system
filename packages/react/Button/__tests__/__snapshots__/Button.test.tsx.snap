// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component testing Button props 1`] = `
<button
  className="c-fEVIFd c-fEVIFd-gwqHNH-size-medium c-fEVIFd-ixhqCZ-appearance-primary"
>
  Button
</button>
`;
